mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/TraceeCmd.scala

```diff
@@ -0,0 +1,92 @@
+package at.forsyte.apalache.tla.tooling.opt
+
+import at.forsyte.apalache.infra.passes.PassChainExecutor
+import at.forsyte.apalache.infra.passes.options._
+import at.forsyte.apalache.io.json.JsonDeserializationError
+import at.forsyte.apalache.io.json.impl.{UJsonRep, UJsonScalaFactory}
+import at.forsyte.apalache.tla.bmcmt.config.TraceeModule
+import org.backuity.clist._
+
+import java.io.File
+import scala.util.{Failure, Success, Try}
+
+/**
+ * This command initiates the 'tracee' command line.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+class TraceeCmd(name: String = "tracee", description: String = "Evaluate expressions over a trace.")
+    extends CheckCmd(name, description) {
+
+  var trace: File = arg[File](description = "a file containing an ITF trace. Must also define --expressions.")
+
+  var expressions: List[String] =
+    arg[List[String]](name = "expressions",
+        description = "TLA+ expressions to be evaluated over a given trace. Must also define --trace.")
+
+  // If we set the --length inside collectTuningOptions it gets overwritten later.
```

Comment:
Yes, Shon and I already had an offline discussion RE options being modifiable mid-run (which they currently aren't). We're probably going to eventually have the passes propagate modules + options.
The use of the persistent length is in `run`:
https://github.com/informalsystems/apalache/blob/4145b809a1ea652d6f2113bf0a76a8ea5c67a09c/mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/TraceeCmd.scala#L75-L76

Reply:
Why do we need a stateful solution that assign `persistentLen` in `collectTuningOptions` tho?

Without knowing the pass executor, it's not obvious that `collectTuningOptions` is called before `run`.

Reply:
So I first tried modifying the `var length` (inherited from `CheckerCmd`) inside `collectTuningOptions`. The problem with that was that one of the option instantiation passes overwrote it again with the default 10. Then I made a class variable `persistentLen`, which remembers the trace length from `collectTuningOptions` and overwrites the `length` in run, called later

Reply:
This is a deficiency of our options architecture, it wasn't really designed to change options mid-run.

Reply:
I like Thomas' reasoning here.

Another option to avoid the mutable state is to call the options pass executor on its own with options set, then change the options as you need for your tracee passchain.

You can see the Shai code for an example of how to call a pass within a command: https://github.com/informalsystems/apalache/blob/e03bc42258aa844869794bf5efc7bb440628db93/shai/src/main/scala/at/forsyte/apalache/shai/v1/CmdExecutorService.scala#L72-L93